test/unit_testing/api_provider_test.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:io';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/http.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:themovietb/src/core/util/data_state.dart';
+import 'package:themovietb/src/data/datasource/remote/movie_api_provider.dart';
+
+class MockClient extends Mock implements Client {}
+
+void main() {
+  late Client client;
+  late MovieApiProvider movieApiProvider;
+  setUpAll(() {
+    client = MockClient();
+    movieApiProvider = MovieApiProvider(client: client);
+    registerFallbackValue(Uri());
+  });
+
+  test(
+    'When client returns a status ok the api provider should return a DataSuccess',
+    () async {
+      when(() => client.get(any())).thenAnswer(
+          (_) => Future.value(Response("{\"status\": \"ok\"}", HttpStatus.ok)));
+      final result = await movieApiProvider.fetchMovieList('asd');
+      expect(result, isInstanceOf<DataSuccess<dynamic>>());
+      verify(() => client.close()).called(1);
```

Comment:
I didn't see the `client.close()` call inside the MovieProvider. Does this test pass?